tests/integration/AutomaticEmails/WooCommerce/Events/PurchasedInCategoryTest.php

```diff
@@ -156,6 +160,51 @@ private function getOrderMock($categories = ['123']) {
     return $orderMock;
   }
 
+  private function _createNewsletter() {
+    $newsletter = Newsletter::createOrUpdate(
+      [
+        'subject' => 'WooCommerce',
+        'preheader' => 'preheader',
+        'type' => Newsletter::TYPE_AUTOMATIC,
+        'status' => Newsletter::STATUS_ACTIVE,
+      ]
+    );
+    $this->_createNewsletterOption(
+      [
+        'sendTo' => 'user',
+        'group' => WooCommerce::SLUG,
+        'event' => PurchasedInCategory::SLUG,
+        'afterTimeType' => 'days',
+        'afterTimeNumber' => 1,
+        'meta' => json_encode(
+          [
+            'option' => [
+              ['id' => '15'],
+            ],
+          ]),
+      ],
+      $newsletter->id
+    );
+    return $newsletter;
+  }
+
+  private function _createSubscriber($customerEmail) {
+    $subscriber = Subscriber::createOrUpdate(Fixtures::get('subscriber_template'));
+    $subscriber->email = $customerEmail;
+    $subscriber->isWoocommerceUser = 1;
+    $subscriber->status = Subscriber::STATUS_SUBSCRIBED;
+    $subscriber->save();
+
+    $subscriberSegment = SubscriberSegment::create();
+    $subscriberSegment->hydrate([
+      'subscriber_id' => $subscriber->id,
+      'segment_id' => Segment::getWooCommerceSegment()->id,
+      'status' => Subscriber::STATUS_SUBSCRIBED,
+    ]);
+    $subscriberSegment->save();
+    return $subscriber;
+  }
```

Comment:
This could be written using Doctrine as well. I think you need only ids in the test.

Reply:
Well, yes, but I didn't write this code, I only moved it around.